packages/js/src/ui/components/Popover.tsx

```diff
@@ -0,0 +1,88 @@
+import { Accessor, createContext, createSignal, JSX, Setter, useContext } from 'solid-js';
+import { useUncontrolledState } from '../helpers';
+import { PopoverContent } from './PopoverContent';
+import { PopoverTarget } from './PopoverTarget';
+
+type Direction = 'top' | 'bottom' | 'left' | 'right';
+type AnchorPosition = 'start' | 'end';
+type Position = Direction | `${Direction}-${AnchorPosition}`;
+type Offset = {
+  x: number | string;
+  y: number | string;
+};
+
+export type PopoverProps = {
+  opened?: boolean;
+  defaultOpened?: boolean;
+  onChange?: (value: boolean) => void;
+  position?: Position;
+  offset?: Offset;
+  children?: JSX.Element;
+};
+
+type PopoverContextType = {
+  targetRef: Accessor<HTMLElement | null>;
+  contentRef: Accessor<HTMLElement | null>;
+  opened: Accessor<boolean>;
```

Comment:
⚠️ Why does this receive an `Accessor`  instead of the value ? Props are reactive.

Reply:
It didnt worked in the context without the accessor.